crates/ethereum-rpc/src/lib.rs

```diff
@@ -556,227 +668,183 @@ fn register_rpc_methods<C: sov_modules_api::Context, Da: DaService>(
+                .await;
+        } else {
+            pending
+                .reject(to_eth_rpc_error("Unsupported subscription topic"))
+                .await;
+        }
+        Ok(())
+    }
 
-                Ok(())
-            },
-        )?;
-
-        rpc.register_subscription(
-            "eth_subscribe",
-            "eth_subscription",
-            "eth_unsubscribe",
-            |parameters, pending, ethereum, _| async move {
-                let mut params = parameters.sequence();
-
-                let topic: String = match params.next() {
-                    Ok(v) => v,
-                    Err(err) => {
-                        pending.reject(err).await;
-                        return Ok(());
-                    }
-                };
-                match topic.as_str() {
-                    "newHeads" => {
-                        let subscription = pending.accept().await.unwrap();
-                        ethereum
-                            .subscription_manager
-                            .as_ref()
-                            .unwrap()
-                            .register_new_heads_subscription(subscription)
-                            .await;
-                    }
-                    "logs" => {
-                        let filter: Filter = match params.next() {
-                            Ok(v) => v,
-                            Err(err) => {
-                                pending.reject(err).await;
-                                return Ok(());
-                            }
-                        };
-                        let subscription = pending.accept().await.unwrap();
-                        ethereum
-                            .subscription_manager
-                            .as_ref()
-                            .unwrap()
-                            .register_new_logs_subscription(filter, subscription)
-                            .await;
-                    }
-                    _ => {
-                        pending
-                            .reject(EthApiError::Unsupported("Unsupported subscription topic"))
-                            .await;
-                        return Ok(());
-                    }
+    async fn subscribe_eth(
+        &self,
+        pending: PendingSubscriptionSink,
+        topic: String,
+        filter: Option<Filter>,
+    ) -> SubscriptionResult {
+        match topic.as_str() {
+            "newHeads" => {
+                let subscription = pending.accept().await?;
+                self.ethereum
+                    .subscription_manager
+                    .as_ref()
+                    .unwrap()
+                    .register_new_heads_subscription(subscription)
+                    .await;
+            }
+            "logs" => {
+                let Some(filter) = filter else {
+                    pending.reject(to_eth_rpc_error("Missing filter")).await;
```

Comment:
you should check what other eth clients return for this scenario. probably the error is standardized